index.bs

```diff
@@ -4445,6 +4493,144 @@ Of the extended attributes defined in this specification, only the [{{Exposed}}]
+the <a href="#es-modules">ECMAScript binding</a>.
+
+Note that like namespaces, modules do not create types.
+
+Of the extended attributes defined in this specification, only the [{{Exposed}}]
+and [{{SecureContext}}] extended attributes are applicable to modules. No
+extended attributes are applicable to [=partial modules=].
+
+[=Modules=] must be annotated with the [{{Exposed}}] [=extended attribute=].
+
+[=Module members=], [=partial modules=], and [=interface members=] of
+[=interfaces=] and [=partial interfaces=] which are [=module members=], must not
+be annotated with the [{{Exposed}}] or [{{SecureContext}}] extended attributes.
+
+Note: This restriction could be relaxed if there is a sufficient use case;
+it is omitted here for simplicity of specification and implementation, and a
+lack of clear use cases. Please <a
+href="https://github.com/heycam/webidl/issues/new?title=Fine%20grained%20exposure%20in%20modules">
+file an issue</a> if you see a need for this feature.
+
+Each module may have <dfn>module evaluation steps</dfn> which are executed
+when the module is first imported. These steps may throw an [=exception=], which
+prevents the module from being accessed.
+
+Rather than defining behavior when accessed, the attributes of a [=module=] act
+as a simple storage location. The steps to "set a module attribute" to an IDL
+value of the type of the attribute. Each [=attribute=] which is a [=module
+member=] must be initialized in [=module evaluation steps=] to set the module
+attribute.
+
+Note: See the ECMAScript binding's definition of [=set a module attribute=].
+
+<div data-fill-with="grammar-Partial"></div>
+
+<div data-fill-with="grammar-PartialDefinition"></div>
+
+<pre class="grammar" id="prod-Module">
+    Module :
+        "module" string "{" ModuleMembers "}" ";"
+</pre>
+
+<pre class="grammar" id="prod-ModuleMembers">
+    ModuleMembers :
+        ExtendedAttributeList ModuleMember ModuleMembers
+        ε
+</pre>
+
+<pre class="grammar" id="prod-ModuleMember">
+    ModuleMember :
+        "interface" InterfaceRest
+        "partial" "interface" InterfaceRest
+        RegularOperation
+        "readonly" AttributeRest
+</pre>
+
+<div class="example">
+
+    The following [=IDL fragment=] which defines a [=module=] exposing a function
+    to get the current timezone, which returns an instance of an interface
+    exposed in the same module.
+
+    <pre class="syntax">
+        [Exposed=Window, SecureContext]
+        module "std:timezone" {
+          interface Timezone {
+            readonly attribute USVString name;
+            long long offsetMs(long long unixTime);
+          };
+          Timezone getCurrentTimezone();
+          readonly attribute Timezone initialTimezone;
+        };
+    </pre>
+
+    <blockquote>
+        The [=module evaluation steps=] of "std:timezone" are as follows:
+            1.  [=set a module attribute|Set the attribute=]
+                "initialTimezone" of "std:timezone" to a [=new=]
```

Comment:
Typographically I think `initialTimezone` should be code and should cross-link to the definition. (Not sure cross-linking is possible/easy with example code though). And "of "std:timezone"" should not be necessary here, ideally, although I'm not sure how to rigorize that.